File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const indexMap = {};
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in indexMap) return [indexMap[nums[i]], i];
        const diff = target - nums[i];
        indexMap[diff] = i;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numsList = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int number = nums[i];
            int difference = target - number;
            
            if (numsList.containsKey(difference)) {
                return new int[] {i, numsList.get(difference)};
            }
            numsList.put(number, i);
        }
        return null;
    }
}